modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/ScriptProcessor.java

```diff
@@ -121,6 +131,15 @@ public ScriptProcessor create(Map<String, Processor.Factory> registry, String pr
                 throw newConfigurationException(TYPE, processorTag, null, "Could not initialize script");
             }
 
+            // compile inline and file-backed scripts to ensure they are valid before constructing the processor.
+            if (script.getType() == ScriptType.INLINE || script.getType() == ScriptType.FILE) {
```

Comment:
I guess my assumption was that stored scripts are compiled when being stored. is this not the case?